packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -973,6 +975,113 @@ function* relevantRemovedRootsFromFields(
 	}
 }
 
+/**
+ * Adds any builds missing from the provided change that are relevant to the change.
+ * Calls {@link relevantRemovedRoots} to determine which builds are relevant.
+ *
+ * @param change - The change with potentially missing builds. Not mutated by this function.
+ * @param getDetachedNode - The function to retrieve a tree chunk from the corresponding detached node id.
+ * @param fieldKinds - The field kinds to delegate to.
+ */
+export function addMissingBuilds(
+	change: TaggedChange<ModularChangeset>,
+	getDetachedNode: (id: DeltaDetachedNodeId) => TreeChunk | undefined,
+	fieldKinds: ReadonlyMap<FieldKindIdentifier, FieldKindWithEditor>,
```

Comment:
Instead of passing `fieldKinds` in, I think it would make for a nicer API to pass in the return value of `relevantRemovedRoots(change, fieldKinds)`.
This makes the function more self-contained and easier to test.
Same for `filterSuperfluousBuilds`